### ISSUE TITLE
vsftpd: kernel complains with KERNEL_NET_NS enabled

### ISSUE THREAD

**Issue by DazzyWalkman** [openwrt/packages#10057]
Environment: OpenWRT snapshot r11035 x86_64, vsftpd-tls 3.0.3-3.

Description: Kernel complains when new connection is made to vsftpd
 (related dmesg output is at the end of this post. There's also a similar output by other user: https://github.com/openwrt/packages/issues/9636).
Nevertheless, vsftpd seems to be functioning normally.

My observation: This is largely dependant on the setting of network namespace support in the kernel config. When the KERNEL_NET_NS is set to enabled, kernel complains.
When I tried to use OpenWRT as an lxc host in Nov. 2018, then lxc in OpenWRT was still experimental, and the required KERNEL_NET_NS was not set by default, I manually enable this feature, and came across this problem for the first time.  I decided not to bring up this issue for it's just a corner case and a customed compiled kernel was needed to reproduce.

Due to recent efforts to make OpenWRT an effective container host with lxc/lxd/docker, KERNEL_NET_NS and some other needed kernel symbols are set by default(https://github.com/openwrt/openwrt/commit/fcb41decf6c622482b20af45a77e62db8d95046e). So users of recent snapshot builds and vsftpd will likely encounter this problem, even if they don't need the container feature.  So now here comes this issue.

My workaround to SILENCE the error message:
add 
`isolate_network=NO`
to vsftpd.conf

Side note: As lxc guest startup/shutdown also printed out similar netns error message back in Nov. 2018, I suspect there's more under the hood and some other packages utilizing network namespace may be affected as well. However, since I've dropped the idea of OpenWRT as lxc host, and I have little knowledge about network namespace, I have no time to set up lxc to see if the same problem persists.  So I would rather file this issue about vsftpd other than report on Flyspray. 

Only patch I found to address newly introduce netns issue: https://github.com/openwrt/openwrt/commit/d344591e72e5ca96a2bf70a2df38961553185ce8
It does not fix the vsftpd issue, of coz. But I think it provides some interesting things about netns.

Another off-topic but remotely related note to vsftpd: seccomp enabled alongside with netns will prevent vsftpd from working out of box with various 500 errors.
Adding 
`seccomp_sandbox=NO`
 to vsftpd.conf
will restore vsftpd to previous behavior.

Be warned, turning off isolate_network and seccomp_sandbox does NOT fix any problem but to make vsftpd behave as if netns and seccomp are disabled, and may have security implications.  It's fine for me to use vsftpd in single user mode on lan with all clients are trusted. 

dmesg output:

```
[49.029228] WARNING: CPU: 0 PID: 7 at __nf_unregister_net_hook+0xe4/0x120
[49.034435] Modules linked in: pppoe ppp_async pppox ppp_generic lzo lz4 iptable_nat ipt_REJECT ipt_MASQUERADE xt_time xt_tcpudp xt_tcpmss xt_statistic xt_state xt_recent xt_nat xt_multiport xt_mark xt_mac xt_limit xt_length xt_hl xt_helper xt_ecn xt_dscp xt_conntrack xt_connmark xt_connlimit xt_connbytes xt_comment xt_TCPMSS xt_REDIRECT xt_NFQUEUE xt_LOG xt_HL xt_FLOWOFFLOAD xt_DSCP xt_CT xt_CLASSIFY slhc r8169 nfnetlink_queue nf_reject_ipv4 nf_nat_ipv4 nf_nat nf_log_ipv4 nf_flow_table_hw nf_flow_table nf_conntrack_rtcache nf_conncount lzo_decompress lzo_compress lz4_decompress lz4_compress kvm irqbypass iptable_raw iptable_mangle iptable_filter ipt_ECN ip_tables e1000e crc_ccitt bnx2 sch_cake nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_tcindex                                           [49.061319]  cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred i2c_dev xt_set ip_set_list_set ip_set_hash_netportnet ip_set_hash_netport ip_set_hash_netnet ip_set_hash_netiface ip_set_hash_net ip_set_hash_mac ip_set_hash_ipportnet ip_set_hash_ipportip ip_set_hash_ipport ip_set_hash_ipmark ip_set_hash_ip ip_set_bitmap_port ip_set_bitmap_ipmac ip_set_bitmap_ip ip_set nfnetlink nf_log_ipv6 nf_log_common ip6table_mangle ip6table_filter ip6_tables ip6t_REJECT x_tables nf_reject_ipv6 igb e1000 ifb vfat fat cifs nls_utf8 nls_iso8859_1 nls_cp950 nls_cp936 nls_cp932 nls_cp437 zram zsmalloc sha512_ssse3 sha512_generic sha256_ssse3 sha256_generic seqiv jitterentropy_rng drbg md5 md4 hmac des_generic ctr cmac ccm arc4 crypto_acompress exfat button_hotplug ptp realtek pps_core libphy                                                          
[49.095519] CPU: 0 PID: 7 Comm: kworker/u2:0 Not tainted 4.19.72 #0
[49.099232] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS rel-1.12.1-0-ga5cab58e9a3f-prebuilt.qemu.org 04/01/2014              
[49.106389] Workqueue: netns cleanup_net
[49.110799] RIP: 0010:__nf_unregister_net_hook+0xe4/0x120
[49.114984] Code: c1 e0 04 49 8d 44 07 08 48 c7 00 e0 9b 6f 81 48 c7 01 80 2f ca 81 41 83 fd 05 75 a9 8b 43 1c 85 c0 75 a2 e8 fe a2 fb ff eb 9b <0f> 0b 48 c7 c7 60 ea 07 82 e8 6e b8 0d 00 eb ba 8b 53 1c 44 89 ee
[49.123826] RSP: 0018:ffffc9000003bd60 EFLAGS: 00010246
[49.127507] RAX: 0000000000000000 RBX: ffffffffa045c040 RCX: 0000000000000000                                                                
[49.131576] RDX: ffff88800744c200 RSI: 0000000000000002 RDI: ffffffff8207ea60                                                                
[49.135702] RBP: ffffc9000003bd88 R08: 0000000000000001 R09: ffff888007d6a840                                                                
[49.139813] R10: 0000000000000000 R11: 0000000000000002 R12: ffff888005b08b80                                                                
[49.143926] R13: 0000000000000002 R14: ffff888005b08000 R15: 0000000000000000                                                                
[49.148023] FS:  0000000000000000(0000) GS:ffff888007800000(0000) knlGS:0000000000000000                                                     
[49.152286] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[49.156071] CR2: 00007f139ac51000 CR3: 00000000060b2000 CR4: 0000000000100ef0                                                                
[49.160222] Call Trace:
[49.163348]  nf_unregister_net_hook+0x20/0x50
[49.166937]  nf_unregister_net_hooks+0x2a/0x40
[49.170446]  0xffffffffa045a045
[49.173708]  ops_exit_list.isra.0+0x2f/0x60
[49.177170]  cleanup_net+0x181/0x250
[49.180546]  process_one_work+0x1ab/0x310
[49.183882]  worker_thread+0x48/0x3f0
[49.187283]  kthread+0x10d/0x130
[49.190481]  ? process_one_work+0x310/0x310
[49.193848]  ? kthread_bind+0x30/0x30
[49.197144]  ret_from_fork+0x35/0x40
[49.201865] ---[ end trace 8ca3e8b085e5aa5c ]---
```

**Comment by schm0** [comment]
Also Off Topic:
vsftpd fails to compile with glibc.
This can be fixed by removing -lnsl from the makefile.
More off topic:
libunwind also fails but I can't remember the exact error message.

**Comment by neheb** [comment]
Is https://sources.debian.org/patches/vsftpd/3.0.3-12/0004-link-local.patch/ relevant?

**Comment by DazzyWalkman** [comment]
@neheb 
I think it irrelevant.
Compiled and tested on x86_64, r12114-943cc81b0f, kernel 4.19.98.
It doesn't hurt to include this patch, but during this test, I found kernel had already stopped complaining, with or without this patch. It's got fixed sometime, somewhere else in OpenWRT code. And I failed to track down what actually is.  
It's probably not an issue about vsftpd.

Vsftpd still doesn't work without `seccomp_sandbox=NO` out of box on OpenWRT. 
Since many packages do not work out of box without proper configuration either,  and I am not sure if it's an intended feature of vsftpd with seccomp enabled, I am closing this issue.
Thank you.
